lib/core/src/chain_swap.rs

```diff
@@ -576,6 +586,11 @@ impl ChainSwapHandler {
                             return Err(anyhow!("Unexpected payload from Boltz status stream"));
                         };
 
+                        if let Err(e) = self.verify_user_lockup_tx(swap).await {
```

Comment:
Do we need to verify the lockup exists for outgoing swap? Why do we care here?

Reply:
Main argument IMO is consistency. We're already doing it when handling `TransactionServerConfirmed` so it made sense to me to also do it when handling `TransactionServerMempool`. But I see your point. The only other reason I can see is debugging. This is unlikely, but it could help detect a bug in our fund-locking logic.

Reply:
I see. The issue is that it will add some delay to the claim and to the whole payment time. Giving this is a chain swap it is not so critical.
So basically it is and extra validation for us to check ourselves. Fine by me.